tests/integration/test_split.py

```diff
@@ -0,0 +1,310 @@
+            num.dsplit(ary, 1)
+        with pytest.raises(expected_exc):
+            np.dsplit(ary, 1)
+
+
+def compare_array(a, b):
+    """
+    Compare two array using zip method.
+    """
+    if len(a) != len(b):
+        return False
+    else:
+        for each in zip(a, b):
+            if not np.array_equal(*each):
+                return False
+    return True
+
+
+def get_indices(arr, axis):
+    indices_arr = []
+
+    if arr.shape[axis] == 1:
+        indices_arr.append(1)  # in index
+
+    elif arr.shape[axis] > 1:
+        for div in range(2, (int)(math.sqrt(arr.shape[axis]) + 1)):
+            if arr.shape[axis] % div == 0:
+                indices_arr.append(div)
+
+                # in index
+                indices_arr.append(list(range(1, arr.shape[axis], div)))
+
+                # out index
+                indices_arr.append(
+                    list(
+                        range(
+                            0,
+                            arr.shape[axis] + div * np.random.randint(1, 10),
+                            div,
+                        )
+                    )
+                )
+                indices_arr.append(
+                    list(
+                        range(
+                            arr.shape[axis] + div * np.random.randint(1, 10),
+                            0,
+                            -div,
+                        )
+                    )
+                )
+
+        # if indivisible
+        if len(indices_arr) == 0:
+            indices_arr.append(1)
+            indices_arr.append(arr.shape[axis])
+            indices_arr.append(list(range(1, arr.shape[axis], 1)))
+
+            # out index
+            indices_arr.append(
+                list(
+                    range(0, arr.shape[axis] + 1 * np.random.randint(1, 10), 1)
+                )
+            )
+            indices_arr.append(
+                list(
+                    range(
+                        arr.shape[axis] + 1 * np.random.randint(1, 10), 0, -1
+                    )
+                )
+            )
+
+    return indices_arr
+
+
+@pytest.mark.parametrize("size", SIZES, ids=str)
+def test_split(size):
+    a = np.random.randint(low=0, high=100, size=size)
+    axis_list = list(range(a.ndim))
+    axis_list.append(-1)
+    for axis in axis_list:
+        input_arr = get_indices(a, axis)
+
+        for input_opt in input_arr:
+            func_num = getattr(num, "split")
+            func_np = getattr(np, "split")
+
+            res_num = func_num(a, input_opt, axis=axis)
+            res_np = func_np(a, input_opt, axis=axis)
+
+            assert compare_array(res_num, res_np)
+
+
+@pytest.mark.parametrize("func_name", ARG_FUNCS)
+@pytest.mark.parametrize("size", SIZES_NO_EMPTY, ids=str)
+def test_split_different_split(size, func_name):
+    a = np.random.randint(low=0, high=100, size=size)
+    axis = AXIS_FUNCS[func_name]
+
+    if axis >= a.ndim:
+        return
```

Comment:
This is a good indication that maybe this test should be split up. This condition means there are "passing" test cases reported in the test output that are actually just no-ops that do nothing and that is a bit misleading. I'd suggest breaking this up into separate test functions for `test_vsplit`, `test_hsplit` and `test_dsplit` and make sure they each have (only) parameterized inputs that are sensible or them individually.

Reply:
fix